ts/main.ts

```diff
@@ -409,3 +466,108 @@ function clearGrid(): void {
+});
+
+for (const categorySet in categorySets) {
+  const gridRow = document.createElement('div');
+  gridRow.classList.add(
+    'flex',
+    'py-1',
+    'ml-2',
+    'rounded-xl',
+    'hover:bg-yellow-200',
+    'main-content',
+    'select-grid',
+  );
+  gridRow.setAttribute('data-grid-ID', categorySet);
+  if (currentGridId === parseInt(categorySet)) {
+    gridRow.classList.add('bg-yellow-500', 'selected-grid-cat-row');
+    gridRow.classList.remove('hover:bg-yellow-200');
+  }
+
+  const imgContainer = document.createElement('div');
+  imgContainer.classList.add('w-1/12', 'main-content');
+  const puzzImg = document.createElement('img');
+  puzzImg.setAttribute('src', './images/puzzle.png');
+  puzzImg.classList.add('main-content');
+  imgContainer.appendChild(puzzImg);
+
+  const gridRowTxt = document.createElement('h1');
+  gridRowTxt.textContent = `Grid #${gridRow.getAttribute('data-grid-ID')}`;
+  gridRowTxt.classList.add('main-content');
+
+  gridRow.appendChild(imgContainer);
+  gridRow.appendChild(gridRowTxt);
+
+  $gridSetsContainer.appendChild(gridRow);
+}
+
+$categoryModal.addEventListener('click', function (event: Event): void {
+  const element = event.target as HTMLElement;
+  const closestCatRow = element.closest('.select-grid') as HTMLElement;
+
+  const $selectedRow = document.querySelector(
+    '.selected-grid-cat-row',
+  ) as HTMLDivElement;
+  if (!$selectedRow) throw new Error('.selected-grid-cat-row query failed!');
+
+  if (
+    !element.classList.contains('main-content') ||
+    closestCatRow.classList.contains('selected-grid-cat-row')
+  ) {
+    $categoryModal.close();
+  } else if (
+    element.tagName === 'IMG' ||
+    element.tagName === 'H1' ||
+    element.tagName === 'DIV'
+  ) {
+    console.log('closestCatRow:', closestCatRow);
+    console.log('selected row:', $selectedRow);
+    const closestGridId = closestCatRow.getAttribute('data-grid-ID') as string;
+    if (!closestGridId) throw new Error('No grid ID found');
+
+    newGridCategories(parseInt(closestGridId));
+
+    $selectedRow.classList.remove('bg-yellow-500', 'selected-grid-cat-row');
+    $selectedRow.classList.add('hover:bg-yellow-200');
+
+    closestCatRow.classList.add('selected-grid-cat-row', 'bg-yellow-500');
+    closestCatRow.classList.remove('hover:bg-yellow-200');
+    $categoryModal.close();
+  }
+});
+
+function newGridCategories(newGridId: number): void {
+  clearGrid();
+  currentGridId = newGridId;
+  $gridIDLabel.textContent = '#' + currentGridId.toString();
+
+  for (let i = 0; i < $topCategories.length; i++) {
+    const topCatElement = $topCategories[i] as HTMLElement;
+    while (topCatElement.firstChild) {
+      topCatElement.removeChild(topCatElement.firstChild);
+    }
+  }
+  for (let i = 0; i < $leftCategories.length; i++) {
+    const leftCatElement = $leftCategories[i] as HTMLElement;
+    while (leftCatElement.firstChild) {
+      leftCatElement.removeChild(leftCatElement.firstChild);
+    }
+  }
```

Comment:
Is there a way that we can put these into the same loop? Your grid is equal side of left and right, correct? You should be able to use one for loop even though you may need to use separate `while` loops.